src/herder/TxSetUtils.h

```diff
@@ -0,0 +1,38 @@
+// Copyright 2022 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#pragma once
+
+#include "util/UnorderedMap.h"
+#include "xdr/Stellar-types.h"
+#include <deque>
+#include <herder/TxSetFrame.h>
+
+namespace stellar
+{
+namespace TxSetUtils
+{
+
+UnorderedMap<AccountID, TxSetFrame::AccountTransactionQueue>
+buildAccountTxQueues(TxSetFrame const& txSet);
```

Comment:
This kind of demonstrates the issue with separating these functions from `TxSetFrame`: on one hand this is an implementation detail and hence shouldn't be exposed in the header (unlike other fucntions here), but since it needs to be shared between these two files, it has to be exposed. I don't think it's a big deal for now, just something to consider.